### PR TITLE
[SOAR-4451]: Lookup Vulnerable Hosts from Teams to use parameters (TRY 2)

### PR DESCRIPTION
**Pull request by jnikitin-r7**
https://issues.corp.rapid7.com/browse/SOAR-4451

## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [x] Workflow verified as working in a test
- [x] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [x] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Workflow includes a detailed Description
- [x] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Triggers, Decisions, and other workflow steps include descriptions where needed
- [x] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [x] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [x] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [x] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

I updated to use workflow parameters for max hosts, max vulns, team name, and channel name.

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

image

image

image

### CODE REVIEW

**Comment by malolans-r7** [comment]
Should we have a consistent description for MS team name and MS channel name parameter? @smorrison-r7

**Comment by jnikitin-r7** [comment]
> Should we have a consistent description for MS team name and MS channel name parameter? @smorrison-r7

I would love something better. I tried to be specific, not "Microsoft Team channel name", but something that tells people that its the specific and only channel name where commands will be enabled in.

**Comment by malolans-r7** [comment]
> > Should we have a consistent description for MS team name and MS channel name parameter? @smorrison-r7
> 
> I would love something better. I tried to be specific, not "Microsoft Team channel name", but something that tells people that its the specific and only channel name where commands will be enabled in.

I used `The Microsoft Teams team name where the workflow will be triggered and respond` and `The Microsoft Teams channel name where the workflow will be triggered and respond`

**Comment by jnikitin-r7** [comment]
> > > Should we have a consistent description for MS team name and MS channel name parameter? @smorrison-r7
> > 
> > 
> > I would love something better. I tried to be specific, not "Microsoft Team channel name", but something that tells people that its the specific and only channel name where commands will be enabled in.
> 
> I used `The Microsoft Teams team name where the workflow will be triggered and respond` and `The Microsoft Teams channel name where the workflow will be triggered and respond`

all set

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md [hunk 1]
```diff
@@ -24,10 +24,15 @@ Sample Microsoft Teams command:
 
 Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
 
-Once the workflow has been imported, **the first Microsoft Teams step will need the team name and channel name updated to suit your Teams environment!** Edit the input with the preset text of `change_me` in first Microsoft Teams step in the workflow. There is also an optional Settings step which can be edited to change the maximum number of vulnerabilities and maximum hosts listed per vulnerability which are returned. The defaults are 50 vulnerabilities and 10 assets per vulnerability.
+This workflow leverages insightConnect's Parameters feature. This feature allows variables used multiple times throughout a workflow to be entered once and then referenced throughout the workflow. To begin, select "Parameters" either from the Workflow home page or from the Builder to begin configuration.
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md:27]:
```suggestion
This workflow leverages InsightConnect's Parameters feature. This feature allows variables used multiple times throughout a workflow to be entered once and then referenced throughout the workflow. To begin, select "Parameters" either from the Workflow home page or from the Builder to begin configuration.
```
Match the above paragraph w/r/t ICON branding

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md [hunk 2]
```diff
@@ -24,10 +24,15 @@ Sample Microsoft Teams command:
...
 
-After configuring the Microsoft Teams and Setting steps, activate the workflow in order to trigger it.
+There are three parameters you will need to configure in order to complete setup of your workflow:
+* `Team Name`:  the Microsoft Teams team name in your environment where the workflow should be triggered and respond
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md:30]:
```suggestion
* `Team Name`: The Microsoft Teams team name in your environment where the workflow should be triggered and respond
```

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md [hunk 3]
```diff
@@ -24,10 +24,15 @@ Sample Microsoft Teams command:
...
+* `Channel Name`:  the Microsoft Teams channel name in your environment where the workflow should be triggered and respond (the channel should exist in the aforementioned team)
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md:31]:
```suggestion
* `Channel Name`: The Microsoft Teams channel name in your environment where the workflow should be triggered and respond (the channel should exist in the aforementioned team)
```

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md [hunk 4]
```diff
@@ -24,10 +24,15 @@ Sample Microsoft Teams command:
...
+* `Max Hosts`: the maximum number of hosts listed per vulnerability (recommended default is 10)
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md:32]:
```suggestion
* `Max Hosts`: The maximum number of hosts listed per vulnerability (recommended default is 10)
```

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md [hunk 5]
```diff
@@ -24,10 +24,15 @@ Sample Microsoft Teams command:
...
+* `Max Vulnerabilities`: the maximum number of returned vulnerabilities (recommended default is 50)
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md:33]:
```suggestion
* `Max Vulnerabilities`: The maximum number of returned vulnerabilities (recommended default is 50)
```

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md [hunk 6]
```diff
@@ -24,10 +24,15 @@ Sample Microsoft Teams command:
...
 
+To configure the parameters, select "Parameters" either from the Workflow Control Panel page or from the Workflow Builder. Once configured, workflow setup is complete.
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md:35]:
```suggestion
To configure the parameters, select "Parameters" either from the Workflow Control Panel page or from the Workflow Builder. Once configured, workflow setup is complete.

```

Comment (mrinehart-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md:35]: Also, this paragraph is a bit redundant with https://github.com/rapid7/insightconnect-workflows/pull/417/files#diff-3049d7ff2243a491195521158c9996cbabb0058048cb6addba1a8f30e23f8b4aR27

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md (hunk 2 above)
Comment (jnikitin-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md]: all set

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md (hunk 3 above)
Comment (jnikitin-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md]: all set

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md (hunk 4 above)
Comment (jnikitin-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md]: all set

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md (hunk 5 above)
Comment (jnikitin-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md]: all set

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md (hunk 1 above)
Comment (jnikitin-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md]: all set

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md (hunk 6 above)
Comment (jnikitin-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md:35]: I believe Charlotte's PR has the same wording? should we just delete this part? if so, I can coordinate with her.

**Review by jschipp-r7 — APPROVED** [review]

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/workflow.spec.yaml [hunk 7]
```diff
@@ -3,13 +3,13 @@ products: ["insightconnect", "insightvm"]
 name: Lookup_Vulnerable_Hosts_From_Teams
 title: Lookup Vulnerable Hosts from Microsoft Teams
 description: Lookup vulnerability details and identify affected hosts with a Microsoft Teams message message trigger.
```
Comment (jnikitin-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Teams/workflow.spec.yaml:5]: @mrinehart-r7 Malo and I noticed this description today? `Microsoft Teams message message trigger`, should we change this to `Microsoft Teams message trigger`?

Comment (mrinehart-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/workflow.spec.yaml:5]: Yes - no need for two messages :)

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md (hunk 6 above)
Comment (jnikitin-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/help.md]: all set, deleted

File: workflows/Lookup_Vulnerable_Hosts_From_Teams/workflow.spec.yaml (hunk 7 above)
Comment (jnikitin-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Teams/workflow.spec.yaml]: fixed, all set

**Review by mrinehart-r7 — APPROVED** [review]

**Review by malolans-r7 — APPROVED** [review]

**Pull request merged**